src/Pagination.tsx

```diff
@@ -0,0 +1,95 @@
+import { Link } from 'react-router-dom';
+import type { PagesData } from './Pagination.types';
+
+//__START_v2
+function getPaginationBaseValue(pagesAll: number, pagesCurrent: number): number {
+    return (Math.floor(pagesCurrent / 10) <= Math.floor(pagesAll / 10)) ?
+        (Math.floor((pagesCurrent - 1) / 10)) :
+        (Math.floor((pagesAll - 1) / 10));
+}
+
+function getPaginationListLength(pagesAll: number, baseValue: number): number {
+    return ((baseValue * 10 + 10) <= pagesAll) ? 10 : pagesAll % 10;
+}
+
+function getPaginationList(pagesAll: number, pagesCurrent: number): number[] {
+    const baseValue = getPaginationBaseValue(pagesAll, pagesCurrent);
+    const paginationListLength = getPaginationListLength(pagesAll, baseValue);
+
+    return Array.from(
+        new Array(paginationListLength), 
+        (_, idx) => baseValue * 10 + idx + 1
+    );
+}
+//__END_v2
```

Comment:
Please search the project for remaining comments by `//` phrase and remove those :)